spyder/plugins/plots/widgets/figurebrowser.py

```diff
@@ -638,36 +677,24 @@ def setup_scrollarea(self):
 
         return self.scrollarea
 
-    def setup_arrow_buttons(self):
-        """
-        Setup the up and down arrow buttons that are placed at the top and
-        bottom of the scrollarea.
-        """
-        # Get the size hint height of the horizontal scrollbar.
-        height = self.scrollarea.horizontalScrollBar().sizeHint().height()
-
-        # Setup the up and down arrow button.
-        up_btn = up_btn = QPushButton(icon=ima.icon('last_edit_location'))
-        up_btn.setFlat(True)
-        up_btn.setFixedHeight(height)
-        up_btn.clicked.connect(self.go_up)
-
-        down_btn = QPushButton(icon=ima.icon('folding.arrow_down_on'))
-        down_btn.setFlat(True)
-        down_btn.setFixedHeight(height)
-        down_btn.clicked.connect(self.go_down)
-
-        return up_btn, down_btn
-
     def set_figureviewer(self, figure_viewer):
         """Set the bamespace for the FigureViewer."""
         self.figure_viewer = figure_viewer
 
     def eventFilter(self, widget, event):
         """
         An event filter to trigger an update of the thumbnails size so that
-        their width fit that of the scrollarea.
+        their width fit that of the scrollarea and to remap some key press
+        events to mimick navigational behaviour of a Qt widget list.
         """
+        if event.type() == QEvent.KeyPress:
+            key = event.key()
+            if key == Qt.Key_Up:
+                self.go_previous_thumbnail()
+                return True
+            elif key == Qt.Key_Down:
+                self.go_next_thumbnail()
+                return True
```

Comment:
Just to check. When this is moved the selection changes and the actual view scrolls?

Reply:
yes